master/buildbot/status/buildrequest.py

```diff
@@ -165,6 +170,10 @@ def asDict(self):
     def asDict_async(self, request=None):
         result = {}
 
+        builder = self.status.getBuilder(self.getBuilderName())
+        if not builder:
+            defer.returnValue(result)
+            return
```

Comment:
I will double check, but buildrequest depends on the builder being present on the master so most of the process wont work, maybe we could figure out this earlier.